src/validation.cpp

```diff
@@ -3430,7 +3430,9 @@ MempoolAcceptResult ChainstateManager::ProcessTransaction(const CTransactionRef&
         state.Invalid(TxValidationResult::TX_NO_MEMPOOL);
         return MempoolAcceptResult::Failure(state);
     }
-    return AcceptToMemoryPool(active_chainstate, *active_chainstate.m_mempool, tx, /* bypass_limits= */ false, test_accept);
+    auto result = AcceptToMemoryPool(active_chainstate, *active_chainstate.m_mempool, tx, /* bypass_limits= */ false, test_accept);
+    active_chainstate.m_mempool->check(active_chainstate.CoinsTip(), active_chainstate.m_chain.Height() + 1);
```

Comment:
In commit "[validation] Always call mempool.check() after processing a new transaction" (f87e07c6fe321f0fb97703c82c0e4122f800589f)

This change does not appear to be a pure refactoring, so it would be helpful if commit description mentioned something about cases when these new checks will be done and whether there are effects on validation or performance.

Reply:
For mainnet nodes with default config, there's no behaviour change. I've added the following text to the commit log:

>    CTxMemPool::check() will carry out internal consistency checks 1/n times,
>    where n is set by the `-checkmempool` configuration option. By default,
>    mempool consistency checks are disabled entirely on mainnet.
>    
>    Therefore, this change has no effect on mainnet nodes running with
>    default configuration. It simply removes the responsibility to trigger
>    mempool consistency checks from net_processing.